File: Sem3/Advanced Methods of Programming/MyInterpreter - JavaFX Graphical Application/project/src/domain/statement/PrintStatement.java
package domain.statement;


import domain.ProgramState;
import domain.adt.MyDictionaryInterface;
import domain.adt.MyListInterface;
import domain.expression.Expression;
import domain.expression.ExpressionException;


public class PrintStatement implements StatementInterface
{
    private Expression expression;


    public PrintStatement(Expression expression)
    {
        this.expression = expression;
    }


    @Override
    public String toString()
    {
        return "print(" + expression.toString() + ")";
    }


    @Override
    public ProgramState execute(ProgramState programState) throws StatementException
    {
        MyListInterface<Integer> queue = programState.getOutput();
        MyDictionaryInterface<String, Integer> symbolTable = programState.getSymbolTable();
        MyDictionaryInterface<Integer, Integer> heapTable = programState.getHeapTable();
        try
        {
            queue.add(expression.eval(symbolTable, heapTable));
        }
        catch (ExpressionException exception)
        {
            throw new StatementException(exception.getMessage() + " Print statement.");
        }
        return null;
    }


    public Expression getExpression()
    {
        return expression;
    }
}

File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/domain/statement/PrintStatement.cs
using System;
using MyInterpreter_CSharp.domain.adt;
using MyInterpreter_CSharp.domain.expression;

namespace MyInterpreter_CSharp.domain.statement
{
    public class PrintStatement : IStatement
    {
        private Expression _expression;

        public PrintStatement(Expression expression)
        {
            _expression = expression;
        }

        public ProgramState Execute(ProgramState programState)
        {
            IMyList<int> queue = programState.Output;
            IMyDictionary<string, int> symbolTable = programState.SymbolTable;
            try
            {
                queue.Add(_expression.Eval(symbolTable));
            }
            catch (ExpressionException exception)
            {
                throw new StatementException(exception.Message + " " + _expression);
            }

            return null;
        }

        public override string ToString()
        {
            return "print(" + _expression + ")";
        }

        public Expression Expression => _expression;
    }
}